### DISCUSSION TITLE
Academy question -> Senior developer certification -> Reporting > Which built-in Pimcore features can be used for reporting?

### DISCUSSION THREAD

**Discussion by friedljoseph43** [pimcore/pimcore#12972] [Q&A / Support]
I don't get it, after studying the chapters and looking through the documentation, I really thought that this was the answer. 
<img width="508" alt="Untitled" src="https://user-images.githubusercontent.com/111962408/186613686-85ca464c-2a7c-4516-9cdd-59f16de6cc29.png">

**Comment by dvesh3** [comment]
Answer:
There are four major ways to integrate reports:

1. **Inline Reports:** Create inline reports within DataObjects e.g. to visualize data quality of an objects. options: C**alculated Values, Dynamic Text Labels**

2. **Custom Reports:** Create centralized reports with different data sources. option: **Custom Reports**

3. **I-frame Reports:** Create custom reports and integrate with the iframe in admin interface. option: **Preview I-Frame Panel**

4. **Custom Portals:** Visual reports and dashboards outside of admin interface implemented on frontend. (this was not in the options though).

**Reply by friedljoseph43** [reply]
Hi @dvesh3,  thank you for your reply.
Though I appreciate your explanation, my question was about question 1. What you're describing is more about the second question which was already correct in my case.

Question 1: Which built-in Pimcore features can be used for reporting?

1. Dynamic Text Labels
2. Object Bricks
3. Calculated Values
4. Preview I-Frame Panel
5. Custom Reports
6. Notification System
7. BI Module
8. Pimcore Assets

**Reply by dvesh3** [reply]
Ok. Could you please explain why you selected _Pimcore Assets_? and not I-Frame preview panel, as this was mentioned in the chapter intro video that this can be used to integrate reports from frontend templates.

**Comment by lamnc-nfq** [comment]
Did you have an answer on it?
I'm trying via the selected but not pass
Preview I-Frame Panel
Dynamic Text Labels
Calculated Values
Custom Reports

**Reply by patriciomg** [reply]
yours are correct.
